src/java.desktop/unix/native/libawt_xawt/awt/screencast_pipewire.c

```diff
@@ -0,0 +1,910 @@
+        return FALSE;
+    }
+
+    pipewire_libhandle = dlopen(VERSIONED_JNI_LIB_NAME("pipewire-0.3", "0"),
+            RTLD_LAZY | RTLD_LOCAL);
+
+    if (!pipewire_libhandle) {
+        DEBUG_SCREENCAST("could not load pipewire library\n", NULL);
+        return FALSE;
+    }
+
+    LOAD_SYMBOL(fp_pw_stream_dequeue_buffer, "pw_stream_dequeue_buffer");
+    LOAD_SYMBOL(fp_pw_stream_state_as_string, "pw_stream_state_as_string");
+    LOAD_SYMBOL(fp_pw_stream_queue_buffer, "pw_stream_queue_buffer");
+    LOAD_SYMBOL(fp_pw_stream_set_active, "pw_stream_set_active");
+    LOAD_SYMBOL(fp_pw_stream_connect, "pw_stream_connect");
+    LOAD_SYMBOL(fp_pw_stream_new, "pw_stream_new");
+    LOAD_SYMBOL(fp_pw_stream_add_listener, "pw_stream_add_listener");
+    LOAD_SYMBOL(fp_pw_stream_disconnect, "pw_stream_disconnect");
+    LOAD_SYMBOL(fp_pw_stream_destroy, "pw_stream_destroy");
+    LOAD_SYMBOL(fp_pw_init, "pw_init");
+    LOAD_SYMBOL(fp_pw_context_connect_fd, "pw_context_connect_fd");
+    LOAD_SYMBOL(fp_pw_core_disconnect, "pw_core_disconnect");
+    LOAD_SYMBOL(fp_pw_context_new, "pw_context_new");
+    LOAD_SYMBOL(fp_pw_thread_loop_new, "pw_thread_loop_new");
+    LOAD_SYMBOL(fp_pw_thread_loop_get_loop, "pw_thread_loop_get_loop");
+    LOAD_SYMBOL(fp_pw_thread_loop_signal, "pw_thread_loop_signal");
+    LOAD_SYMBOL(fp_pw_thread_loop_wait, "pw_thread_loop_wait");
+    LOAD_SYMBOL(fp_pw_thread_loop_accept, "pw_thread_loop_accept");
+    LOAD_SYMBOL(fp_pw_thread_loop_start, "pw_thread_loop_start");
+    LOAD_SYMBOL(fp_pw_thread_loop_stop, "pw_thread_loop_stop");
+    LOAD_SYMBOL(fp_pw_thread_loop_destroy, "pw_thread_loop_destroy");
+    LOAD_SYMBOL(fp_pw_thread_loop_lock, "pw_thread_loop_lock");
+    LOAD_SYMBOL(fp_pw_thread_loop_unlock, "pw_thread_loop_unlock");
+    LOAD_SYMBOL(fp_pw_properties_new, "pw_properties_new");
+
+    return TRUE;
+
+    fail:
+        dlclose(pipewire_libhandle);
+        pipewire_libhandle = NULL;
+    return FALSE;
+}
+
+void storeRestoreToken(const gchar* oldToken, const gchar* newToken) {
+
+    JNIEnv* env = (JNIEnv *) JNU_GetEnv(jvm, JNI_VERSION_1_2);
+    DEBUG_SCREENCAST("saving token, old: |%s| > new: |%s|\n", oldToken, newToken);
+    if (env) {
+        jstring jOldToken = NULL;
+        if (oldToken) {
+            jOldToken = (*env)->NewStringUTF(env, oldToken);
+            EXCEPTION_CHECK_DESCRIBE();
+            if (!jOldToken) {
+                return;
+            }
+        }
+        jstring jNewToken = (*env)->NewStringUTF(env, newToken);
+        EXCEPTION_CHECK_DESCRIBE();
+        if (!jNewToken) {
+            (*env)->DeleteLocalRef(env, jOldToken);
+            return;
+        }
+
+        jintArray allowedBounds = NULL;
+        if (screenSpace.screenCount > 0) {
+            allowedBounds = (*env)->NewIntArray(env, screenSpace.screenCount*4);
+            EXCEPTION_CHECK_DESCRIBE();
+            if (!allowedBounds) {
+                (*env)->ExceptionClear(env);
```

Comment:
I guess as mentioned above, DESCRIBE clears exception also so no need of calling  explicit Clear
https://docs.oracle.com/en/java/javase/17/docs/specs/jni/functions.html#exceptiondescribe